File: app/src/main/java/com/univlyon1/tools/agenda/SearchRoomFragment.java
package com.univlyon1.tools.agenda;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CalendarView;
import android.widget.EditText;
import androidx.fragment.app.Fragment;
import com.iutcalendar.calendrier.CurrentDate;


public class SearchRoomFragment extends Fragment {

    private Button searchBtn, resetBtn;

    private EditText timeChooser;
    private CalendarView calendarView;
    private View view;

    public SearchRoomFragment() {
    }

    private void initVariables() {
        searchBtn = view.findViewById(R.id.searchBtn);
        resetBtn = view.findViewById(R.id.resetBtn);

        timeChooser = view.findViewById(R.id.editTextTime);
        calendarView = view.findViewById(R.id.calendarView);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.fragment_search_room, container, false);

        initVariables();

        resetValues();

        resetBtn.setOnClickListener((view) -> resetValues());

        return view;
    }

    private void resetValues() {
        timeChooser.setText(new CurrentDate().timeToString());

        calendarView.setDate(new CurrentDate().getTimeInMillis());
    }
}

File: app/src/main/java/com/iutcalendar/search/fragment/agenda/SearchRoomFragment.kt
package com.iutcalendar.search.fragment.agenda

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import androidx.navigation.Navigation
import com.iutcalendar.calendrier.CurrentDate
import com.iutcalendar.calendrier.DateCalendrier
import com.iutcalendar.calendrier.SearchCalendrier
import com.iutcalendar.data.DataGlobal
import com.iutcalendar.dialog.DialogMessage
import com.iutcalendar.filedownload.FileDownload
import com.iutcalendar.fragment.NavigatorManager
import com.iutcalendar.settings.SettingsApp
import com.iutcalendar.settings.URLSetterFragment
import com.univlyon1.tools.agenda.R
import com.univlyon1.tools.agenda.databinding.FragmentSearchRoomBinding
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import java.util.*


class SearchRoomFragment : Fragment() {
    private lateinit var binding: FragmentSearchRoomBinding

    private var dateSelected = DateCalendrier()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_search_room, container, false).also { view ->
            binding = FragmentSearchRoomBinding.bind(view)

            resetValues()
            binding.resetBtn.setOnClickListener { resetValues() }
            binding.searchBtn.setOnClickListener {
                val bundle = bundleOf("timeInMillis" to dateSelected.timeInMillis)
                Navigation.findNavController(view).navigate(R.id.goToDisplay, bundle)
            }


            DataGlobal.getSavedRoomsPath(requireContext()).let { path ->
                lifecycleScope.launch(Dispatchers.IO) {
                    if (path.isNullOrBlank() || !FileDownload.isValideURL(path)) {
                        withContext(Dispatchers.Main) {
                            DialogMessage.showWarning(
                                context, getString(R.string.url_rooms_invalide), getString(R.string.url_rooms_invalide_msg)
                            ) {
                                NavigatorManager.startFragmentFromFragment(this@SearchRoomFragment, URLSetterFragment())
                            }
                        }

                    } else {
                        SearchCalendrier.loadCalendrierRoom(requireContext())
                        withContext(Dispatchers.Main) {
                            loading(false)
                        }
                    }
                }
            }
        }
    }

    private fun loading(state: Boolean) {
        if (state) {
            binding.blockClickView.visibility = View.VISIBLE
            binding.progressBar.visibility = View.VISIBLE
        } else {
            binding.blockClickView.visibility = View.INVISIBLE
            binding.progressBar.visibility = View.INVISIBLE
        }
    }

    private fun resetValues() {
        binding.editTime.apply {
            setIs24HourView(SettingsApp.locale == SettingsApp.localFrance)

            CurrentDate().let {
                hour = it.hour
                minute = it.minute
            }
            setOnTimeChangedListener { _, hourOfDay, minuteTime ->
                dateSelected.hour = hourOfDay
                dateSelected.minute = minuteTime
            }
        }

        binding.calendarView.date = System.currentTimeMillis()
        dateSelected = CurrentDate()
        binding.calendarView.setOnDateChangeListener { _, year, month, dayOfMonth ->
            dateSelected.from(GregorianCalendar(year, month, dayOfMonth, binding.editTime.hour, binding.editTime.minute, 0))
        }
    }
}